easybuild/framework/easyblock.py

```diff
@@ -2202,8 +2202,30 @@ def make_module_step(self, fake=False):
                 for line in txt.split('\n'):
                     self.dry_run_msg(' ' * 4 + line)
         else:
-            write_file(mod_filepath, txt)
-            self.log.info("Module file %s written: %s", mod_filepath, txt)
+            module_only = build_option('module_only')
+            backup_modules = build_option('backup_modules')
+            if module_only and os.path.exists(mod_filepath) and backup_modules and not fake:
+                warning_msg = "Old module file found. Backing it up in %s."
+                hidden = False
+                if isinstance(self.module_generator, ModuleGeneratorTcl):
+                     hidden = True
+                else:
+                    hidden = False
+                mod_bck_filepath = back_up_file(mod_filepath, backup_extension="bck", hidden=hidden)
+                write_file(mod_filepath, txt)
+                # diff will return 1 if there are differences. We have to set log_ok and log_all to False to ignore the
+                # return code
+                (mod_diff, _) = run_cmd("diff -u %s %s" % (mod_bck_filepath, mod_filepath), log_ok=False, log_all=False)
+                if mod_diff:
+                    self.log.info(warning_msg + ' Diff is:\n%s', mod_bck_filepath, mod_diff)
+                    print_warning(warning_msg % mod_bck_filepath + ' Diff is:\n%s' % mod_diff)
+                else:
+                    self.log.info(warning_msg + ' No differences found.', mod_bck_filepath)
+                    print_warning(warning_msg % mod_bck_filepath + ' No differences found.')
+                self.log.info("Module file %s written: %s", mod_filepath, txt)
+            else:
+                write_file(mod_filepath, txt)
+                self.log.info("Module file %s written: %s", mod_filepath, txt)
```

Comment:
This is there twice now, why not just do this after backing up?

You would need delay the diff'ing a bit...